### PR TITLE
Remove _.isInteger

### PR DESCRIPTION
**Pull request by nvdai2401**
Addresses: #7747

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js [hunk 1]
```diff
@@ -120,7 +120,7 @@ module.exports = {
     if (apiGateway.minimumCompressionSize) {
       const minimumCompressionSize = apiGateway.minimumCompressionSize;
 
-      if (!_.isInteger(minimumCompressionSize)) {
+      if (!Number(minimumCompressionSize)) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js:123]: We should remove this condition entirely (note that if value is not an integer, at this point it'll be neatly reported by validation on AWS CloudFormation side, and we also have shortly coming schema based validation https://github.com/serverless/serverless/issues/6562 that will further improve that experience)

File: lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js [hunk 2]
```diff
@@ -395,7 +395,7 @@ module.exports = {
         cors.methods.push(http.method.toUpperCase());
       }
       if (_.has(cors, 'maxAge')) {
-        if (!_.isInteger(cors.maxAge) || cors.maxAge < 1) {
+        if (!Number(cors.maxAge) || cors.maxAge < 1) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js:398]: Note that `Number` does coercion to number, so this is useless in this condition. Let's just keep it as `if (cors.maxAge < 1)`

File: lib/plugins/aws/package/compile/functions/index.js [hunk 3]
```diff
@@ -400,7 +400,7 @@ class AwsCompileFunctions {
         // Try convert reservedConcurrency to integer
         const reservedConcurrency = Number(functionObject.reservedConcurrency);
 
-        if (_.isInteger(reservedConcurrency)) {
+        if (typeof reservedConcurrency === 'number' && !isNaN(reservedConcurrency)) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.js:403]: Just use `isNaN` check (note that value is guraanteed there to be type of `number` as line above coercion to number takes place)

File: lib/plugins/aws/package/compile/functions/index.js [hunk 4]
```diff
@@ -518,7 +518,7 @@ class AwsCompileFunctions {
 
           const provisionedConcurrency = Number(functionObject.provisionedConcurrency);
 
-          if (!_.isInteger(provisionedConcurrency)) {
+          if (typeof provisionedConcurrency !== 'number') {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.js:521]: Let's remove this condition entirely

File: lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js (hunk 1 above)
Comment (nvdai2401) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js:123]: @medikoo There is a fail test case in ```restApi.test.js``` when I remove this condition entirely. Should I remove that test case?
This is the error:
> 4) #compileRestApi()
       should throw error if minimumCompressionSize is not an integer:
     AssertionError: expected promise to be rejected with 'Error' but it was fulfilled with undefined
      at Proxy.<anonymous> (node_modules\chai-as-promised\lib\chai-as-promised.js:169:53)
      at Proxy.<anonymous> (node_modules\chai-as-promised\lib\chai-as-promised.js:47:29)
      at Proxy.methodWrapper (node_modules\chai\lib\chai\utils\addMethod.js:57:25)
      at Context.it (lib\plugins\aws\package\compile\events\apiGateway\lib\restApi.test.js:223:64)
      at process._tickCallback (internal/process/next_tick.js:61:11)

Comment (medikoo) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/restApi.js:123]: @nvdai2401 yes let's remove this test.

**Comment by nvdai2401** [comment]
@medikoo I've updated my PR. Please check it again!

File: lib/plugins/aws/package/compile/functions/index.js [hunk 5]
```diff
@@ -400,7 +400,7 @@ class AwsCompileFunctions {
         // Try convert reservedConcurrency to integer
         const reservedConcurrency = Number(functionObject.reservedConcurrency);
 
-        if (_.isInteger(reservedConcurrency)) {
+        if (!isNaN(reservedConcurrency)) {
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.js:403]: Let's actually remove this condition (assign value unconditionally)

File: lib/plugins/aws/package/compile/functions/index.js [hunk 6]
```diff
@@ -518,12 +518,12 @@ class AwsCompileFunctions {
...
-              'You should use integer as provisionedConcurrency value on function: ' +
-                `${functionResource.Properties.FunctionName}`
-            );
-          }
+          // if (typeof provisionedConcurrency !== 'number') {
+          //   throw new this.serverless.classes.Error(
+          //     'You should use integer as provisionedConcurrency value on function: ' +
+          //       `${functionResource.Properties.FunctionName}`
+          //   );
+          // }
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/functions/index.js:526]: Let's remove this block (we should not commit in commented out code)

File: lib/plugins/aws/package/compile/functions/index.js (hunk 5 above)
Comment (nvdai2401) [reply lib/plugins/aws/package/compile/functions/index.js:403]: @medikoo when I assign value unconditionally, I get a fail test case inde ```index.test.js```. Should I remove this test case?
This test case:

> 4) AwsCompileFunctions
       #compileFunctions()
         should throw an informative error message if non-integer reserved concurrency limit set on function:
     AssertionError: expected promise to be rejected with an error including 'You should use integer as reservedConcurrency value on function: new-service-dev-func' but it was fulfilled with [ 'func' ]
      at Proxy.<anonymous> (node_modules\chai-as-promised\lib\chai-as-promised.js:169:53)
      at Proxy.<anonymous> (node_modules\chai-as-promised\lib\chai-as-promised.js:47:29)
      at Proxy.methodWrapper (node_modules\chai\lib\chai\utils\addMethod.js:57:25)
      at Context.<anonymous> (lib\plugins\aws\package\compile\functions\index.test.js:2390:69)
      at processTicksAndRejections (internal/process/task_queues.js:76:11)

Comment (medikoo) [reply lib/plugins/aws/package/compile/functions/index.js:403]: Yes let's remove this test case

**Comment by nvdai2401** [comment]
@medikoo I've updated my PR. Please check it again!

**Pull request merged**